### DISCUSSION TITLE
Rilevare stato relé

### DISCUSSION THREAD

**Discussion by NiKo-OnGit** [indomus/forum#2979] [GENERALE - Discussioni generiche]
Ciao a tutti,
sto cercando di integrare in HA un allarme vecchio stile che non ha nessuna connessione per la rete.
L'idea è quella di usare uno shelly/sonoff collegando il contatto pulito all'ingresso chiave per armare/disarmare l'allarme, ma avrei il problema di non avere modo di sapere lo stato attuale dell'allarme.
Nella morsettiera dell'allarme sono presenti due relé che possono essere programmati per attivarsi al momento dell'inserimento/disinserimento (da usare per collegare ad esempio un led spia), quindi pensavo si potessero sfruttare in qualche modo.
Quali dispositivi esistono (compatibili con HA ovviamente) che possano indicare lo stato di un relé?

**Comment by jetscrubber** [comment]
Ciao, anch'io ero nella tua stessa situazione ed ho risolto collegando alla morsettiera dell'allarme (che nel mio caso è un contatto pulito) un sensore di allagamento zigbee. Quando inserisco l'allarme il contatto si chiude e il sensore mi segnala l'allagamento mentre quando l'allarme è disinserito il sensore mi segnala asciutto. Poi in home assistant gestisco i due stati dell'entità come voglio.

**Reply by dante70** [reply]
Ciao,credo che al caso tuo siano adatti ad esempio  gli shelly i4 che hanno 4 ingressi che vengono esposti in ha.Se invece vuoi fare in autonomia e usi esphome prendi un D1 mini e lo programmi  collegando i relativi gpio ali relè della scheda allarme.Se volessi quest'ultima soluzione poi possiamo approfondire

**Reply by jetscrubber** [reply]
Ciao Dante, io avrei il problema di rilevare la presenza o meno di tensione (230v) su un contatto. Anche in questo caso si potrebbe usare un esp32? E come? Di certo non posso andare diretto sui gpio con quella tensione.

**Reply by dante70** [reply]
Al momento mi vengono solo 2 soluzioni:
1) metti un relè 220 che chiude contatto e quindi manda ad esempio a massa un gpio dell'esp32.
2) compri una schedina di interfaccia che ho visto su AliExpress che fa la stessa cosa del relè ed e' isolata con foto accoppiatore.Non ho capito se però possa funzionare a 5v
